classes/plugins/PKPPubIdPluginHelper.inc.php

```diff
@@ -195,11 +195,9 @@ function clearPubId($contextId, $pubIdPlugInClassName, $pubObject) {
 				if (get_class($pubIdPlugin) == $pubIdPlugInClassName) {
 					// clear the pubId:
 					// delete the pubId from the DB
-					$pubObjectType = $pubIdPlugin->getPubObjectType($pubObject);
-					$daos = $pubIdPlugin->getDAOs();
-					$dao = $daos[$pubObjectType];
+					$dao = $pubObject->getDAO();
 					$pubObjectId = $pubObject->getId();
-					if ($pubObjectType == 'SubmissionFile') {
+					if (is_a($pubObject, 'SubmissionFile')) {
```

Comment:
(I've merged the PRs, but think this is a worthwhile tweak for separate submission. Mind tinkering with it, or if you're working on other stuff, filing it for later?)